src/connection.ts

```diff
@@ -746,31 +746,31 @@ export class Connection extends EventEmitter {
       return
     }
     if (this.remoteClosed) {
-      this.debug('remote connection is already closed, not starting another loop')
+      this.log.debug('remote connection is already closed, not starting another loop')
       this.safeEmit('_send_loop_finished')
       return
     }
     if (!this.destinationAccount) {
-      this.debug('not sending because we do not know the client\'s address')
+      this.log.debug('not sending because we do not know the client\'s address')
       this.sending = false
       return
     }
 
     this.sending = true
-    this.debug('starting send loop')
+    this.log.debug('starting send loop')
 
     try {
       while (this.sending) {
         // Send a test packet first to determine the exchange rate
         if (!this.exchangeRate) {
-          this.debug('determining exchange rate')
+          this.log.trace('determining exchange rate')
           await this.sendTestPacket()
 
           if (this.exchangeRate) {
             this.safeEmit('connect')
-            this.debug('connected')
+            this.log.trace('connected')
           } else {
-            this.debug('unable to determine exchange rate')
+            this.log.debug('unable to determine exchange rate')
```

Comment:
Probably better to use error here since if the exchange rate cannot be determined that should be surfaced.